packages/screens/Rakki/components/BuyTickets/BuyTicketsModal.tsx

```diff
@@ -0,0 +1,335 @@
+import { useQueryClient } from "@tanstack/react-query";
+import Long from "long";
+import { Dispatch, FC, SetStateAction, useEffect, useState } from "react";
+import { TextInput, TextStyle, View } from "react-native";
+
+import rakkiTicketSVG from "@/assets/icons/rakki-ticket.svg";
+import { BrandText } from "@/components/BrandText";
+import { SVG } from "@/components/SVG";
+import { Box } from "@/components/boxes/Box";
+import { PrimaryButton } from "@/components/buttons/PrimaryButton";
+import { SecondaryButton } from "@/components/buttons/SecondaryButton";
+import { MainConnectWalletButton } from "@/components/connectWallet/MainConnectWalletButton";
+import { GradientText } from "@/components/gradientText";
+import ModalBase from "@/components/modals/ModalBase";
+import { useFeedbacks } from "@/context/FeedbacksProvider";
+import { Info, RakkiClient } from "@/contracts-clients/rakki";
+import { useBalances } from "@/hooks/useBalances";
+import useSelectedWallet from "@/hooks/useSelectedWallet";
+import { getNetworkFeature, NetworkFeature } from "@/networks";
+import { getKeplrSigningCosmWasmClient } from "@/networks/signer";
+import { ModalTicketImage } from "@/screens/Rakki/components/BuyTickets/ModalTicketImage";
+import { prettyPrice } from "@/utils/coins";
+import {
+  errorColor,
+  neutral00,
+  neutral17,
+  neutral22,
+  neutral33,
+  neutral77,
+  neutralA3,
+  neutralFF,
+} from "@/utils/style/colors";
+import {
+  fontSemibold13,
+  fontSemibold14,
+  fontSemibold16,
+} from "@/utils/style/fonts";
+import { layout } from "@/utils/style/layout";
+import { modalMarginPadding } from "@/utils/style/modals";
+
+export const BuyTicketsModal: FC<{
+  visible: boolean;
+  setModalVisible: Dispatch<SetStateAction<boolean>>;
+  info: Info;
+  networkId: string;
+}> = ({ visible, setModalVisible, info, networkId }) => {
+  const selectedWallet = useSelectedWallet();
+  const remainingTickets = info.config.max_tickets - info.current_tickets_count;
+  const [ticketAmount, setTicketAmount] = useState("1");
```

Comment:
IMHO, if possible, we should use tickerAmount as number format , it's not a "big" number so we should process its value from the input to have the number format instead of string which is quite confusing

Reply:
Sorry for adding these changes..
I didn't modify that is this PR, just splitted code from `RakkiScreen.tsx`.
This piece of code is the same as on `origin/main`: https://github.com/TERITORI/teritori-dapp/blob/main/packages/screens/Rakki/RakkiScreen.tsx#L119

Reply:
Same answer for this comment: https://github.com/TERITORI/teritori-dapp/pull/1434#discussion_r1898942611